obspy/clients/eida/__init__.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+FDSN Web Service client with EIDA routing and authentication support.
+
+:copyright:
+    The ObsPy Development Team (devs@obspy.org)
+    Helmholtz-Zentrum Potsdam - Deutsches GeoForschungsZentrum GFZ
+    (geofon@gfz-potsdam.de)
+:license:
+    GNU Lesser General Public License, Version 3
+    (https://www.gnu.org/copyleft/lesser.html)
+"""
+import io
+import obspy.clients.fdsn
+from . import fetch
+
+try:
+    # Python 2.x
+    import urlparse
+
+except ImportError:
+    # Python 3.x
+    import urllib.parse as urlparse
+
+class Client(obspy.clients.fdsn.Client):
+    def __init__(self, base_url='GFZ', retry_count=10, retry_wait=60, maxthreads=5, credentials=None, authdata=None, **kwargs):
+        """
+        Initializes an FDSN/EIDA Web Service client.
+
+        :type base_url: str
+        :param base_url: Base URL of FDSN/EIDA web service compatible server
+            (e.g. "http://geofon.gfz-potsdam.de") or key string for recognized
+            server.
+        :type debug: bool
+        :param debug: Debug flag.
+        :type timeout: float
+        :param timeout: Maximum time (in seconds) to wait for a single request
+            to receive the first byte of the response (after which an exception
+            is raised).
+        :type retry_count: int
+        :param retry_count: Number of retries.
+        :type retry_wait: int
+        :param retry_wait: Seconds to wait before each retry.
+        :type maxthreads: int
+        :param maxthreads: Maximum number of download threads.
+        :type credentials: dict
+        :param credentials: url -> (username, password).
+        :type authdata: string
+        :param authdata: Authentication token (PGP base64 format).
+
+        """
+        obspy.clients.fdsn.Client.__init__(self, base_url, **kwargs)
+        self.__retry_count = retry_count
+        self.__retry_wait = retry_wait
+        self.__maxthreads = maxthreads
+        self.__credentials = credentials
+        self.__authdata = authdata
+
+    def _create_url_from_parameters(self, service, *args):
+        if service in ('dataselect', 'station'):
+            # construct a URL for the routing service
+            u = urlparse.urlparse(obspy.clients.fdsn.Client._create_url_from_parameters(self, service, *args))
+            return urlparse.urlunparse((u.scheme, u.netloc, u'/eidaws/routing/1/query', u'', u.query + u'&service=' + service, u''))
+
+        else: # 'event' is not routed
+            return obspy.clients.fdsn.Client._create_url_from_parameters(self, service, *args)
```

Comment:
Since inheritance is declared with the class definition this would be cleaner with `super(Client, self)._create_url_from_parameters(service, *args)`, I think.